controllers/parse/parser.go

```diff
@@ -0,0 +1,38 @@
+package parse
+
+import "errors"
+
+type Parser struct {
```

Comment:
How about the following implementations:

```go
type Property struct {
	Name  string
	Value interface{}
}

type Properties []*Property

func (ps Properties) New(Name string, Value interface{}) *Property {
	p := &Property{
		Name:  Name,
		Value: Value,
	}

	ps = append(ps, p)

	return p
}

func (ps Properties) Update(Name string, Value interface{}) {
	for _, p := range ps {
		if p.Name == Name {
			p.Value = Value
			return
		}
	}
}

func (p *Property) ToInt() int {
	if p.Value != nil {
		if v, ok := p.Value.(int); ok {
			return v
		}
	}

	return 0
}

func (p *Property) ToString() string {
	if p.Value != nil {
		if v, ok := p.Value.(string); ok {
			return v
		}
	}

	return ""
}

// Add more ToXXX when necessary
```
The property supports kinds of `ToXXX()` helper functions for getting the value. Which func should be called depending on the named property that is known by the caller.